### PR TITLE
Terminal animation added

### PR DESCRIPTION
**Pull request by sourav-sharma7**
Signed-off-by: sourav-sharma7 <souravsharma1321@gmail.com>

**Description**
Animated terminal for meshery docs is added

This PR fixes #734  

**Notes for Reviewers**

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by sourav-sharma7** [comment]
This PR fixed issue#734 for the meshery.io repository, terminal animation is added.

**Comment by leecalcote** [comment]
@adithyaakrishna what do you think?

**Comment by Santosh1176** [comment]
> @Santosh1176, what do you think?

Looks great. A couple of updates to this could bring more value to the UX.
1. The width of the terminal could be increased to fit in the complete command.
2. Currently, the animation runs on all three terminals at once, while two of them are out of viewport and invisible. If feasible, the animation for the other two terminals could be triggered when those enter the viewport.

**Comment by sourav-sharma7** [comment]
@Santosh1176 
Great suggestions will try to implement them as well

**Comment by leecalcote** [comment]
Thank you, @Santosh1176 👍

**Comment by l5io** [comment]
🚀 Preview for commit 35796acb7a9c28acc65af59b23e8c0f4a68edd43 at: https://62e3b58992136d73617cfa0a--meshery-docs-preview.netlify.app

File: docs/assets/js/terminal.js [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+ * async/await.
+ *
+ * @author Ines Montani <ines@ines.io>
+ * @version 0.0.1
+ * @license MIT
+ */
+
+ 'use strict';
+
+//////array of terminal count
```
Comment (adithyaakrishna) [inline docs/assets/js/terminal.js:13]:
```suggestion
// Array of terminal count
```

File: docs/assets/js/terminal.js [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+
+const animatedTerminal = [];
+
+/////function to check if element is in view
```
Comment (adithyaakrishna) [inline docs/assets/js/terminal.js:17]:
```suggestion
// Function to check if element is in view
```

File: docs/assets/js/terminal.js [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+         this.container.setAttribute('data-termynal', '');
+         this.container.innerHTML = '';
+         var divIndex = this.container.id;
+         var self = this;
+
+         if(inView(divIndex)){
+            animatedTerminal.push(divIndex);
+            self.start();
+         }
+         /////scroll eventlistiner for mobile view
```
Comment (adithyaakrishna) [inline docs/assets/js/terminal.js:103]:
```suggestion
         // scroll eventlistiner for mobile view
```

File: docs/assets/js/terminal.js [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+         $(document.body).on('touchmove', mobilescroll);
+         ////mobile view function
```
Comment (adithyaakrishna) [inline docs/assets/js/terminal.js:105]:
```suggestion
         // mobile view function
```

Comment (adithyaakrishna) [inline docs/assets/js/terminal.js:105]:
```suggestion
         // mobile view function
```

Comment (adithyaakrishna) [inline docs/assets/js/terminal.js:105]:
```suggestion
         // mobile view function
```

File: docs/assets/js/terminal.js [hunk 5]
```diff
@@ -0,0 +1,251 @@
...
+         function mobilescroll(){ 
+            
+            if (inView(divIndex) && !animatedTerminal.includes(divIndex)) {
+                animatedTerminal.push(divIndex);
+                self.start();    
+                }
+        } 
+
+         /////scroll eventlistiner function for window view
```
Comment (adithyaakrishna) [inline docs/assets/js/terminal.js:114]:
```suggestion
         // scroll eventlistiner function for window view
```

**Review by adithyaakrishna — CHANGES_REQUESTED** [review]
@sourav-sharma7 Could you make the following changes? Once done, I will go ahead and merge this :)

**Review by adithyaakrishna — APPROVED** [review]
Added the changes and updated the branch 🚀

**Comment by l5io** [comment]
🚀 Preview for commit 086e78fbc5c0d9e6d00d8c87dd150aee170950b6 at: https://62ee2718ac35f82737d93a5b--meshery-docs-preview.netlify.app

**Comment by adithyaakrishna** [comment]
This PR is wrt https://github.com/meshery/meshery.io/issues/734 and https://github.com/meshery/meshery.io/issues/711

**Pull request merged**